crates/sui-types/src/transaction.rs

```diff
@@ -1010,75 +1018,139 @@ impl Display for ProgrammableMoveCall {
+                write!(f, "\n └")
             }
             Command::TransferObjects(objs, addr) => {
-                write!(f, "TransferObjects([")?;
-                write_sep(f, objs, ",")?;
-                write!(f, "],{addr})")
+                write!(f, "TransferObjects:\n ┌\n │ Arguments: \n │   ")?;
+                write_sep(f, objs, "\n │   ")?;
+                write!(f, "\n │ Address: {addr}\n └")
             }
             Command::SplitCoins(coin, amounts) => {
-                write!(f, "SplitCoins({coin}")?;
-                write_sep(f, amounts, ",")?;
-                write!(f, ")")
+                write!(f, "SplitCoins:\n ┌\n │ Coin: {coin}\n │ Amounts: \n │   ")?;
+                write_sep(f, amounts, "\n │   ")?;
+                write!(f, "\n └")
             }
             Command::MergeCoins(target, coins) => {
-                write!(f, "MergeCoins({target},")?;
-                write_sep(f, coins, ",")?;
-                write!(f, ")")
+                write!(f, "MergeCoins:\n ┌\n │ Target: {target}\n │ Coins: \n │   ")?;
+                write_sep(f, coins, "\n │   ")?;
+                write!(f, "\n └")
             }
             Command::Publish(_bytes, deps) => {
-                write!(f, "Publish(_,")?;
-                write_sep(f, deps, ",")?;
-                write!(f, ")")
+                write!(f, "Publish:\n ┌\n │ Dependencies: \n │   ")?;
+                write_sep(f, deps, "\n │   ")?;
+                write!(f, "\n └")
             }
             Command::Upgrade(_bytes, deps, current_package_id, ticket) => {
-                write!(f, "Upgrade(_,")?;
-                write_sep(f, deps, ",")?;
-                write!(f, ", {current_package_id}")?;
-                write!(f, ", {ticket}")?;
-                write!(f, ")")
+                write!(f, "Upgrade:\n ┌\n │ Dependencies: \n │   ")?;
+                write_sep(f, deps, "\n │   ")?;
+                write!(f, "\n │ Current Package ID: {current_package_id}")?;
+                write!(f, "\n │ Ticket: {ticket}")?;
+                write!(f, "\n └")
             }
         }
     }
 }
 
 impl Display for ProgrammableTransaction {
     fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        let mut builder = TableBuilder::default();
+
         let ProgrammableTransaction { inputs, commands } = self;
-        writeln!(f, "Inputs: {inputs:?}")?;
-        writeln!(f, "Commands: [")?;
-        for c in commands {
-            writeln!(f, "  {c},")?;
+
+        for (i, input) in inputs.iter().enumerate() {
+            match input {
+                Pure(v) => {
+                    if v.len() <= 16 {
+                        builder.push_record(vec![format!("{i:<3} Pure Arg          {:?}", v)]);
+                    } else {
+                        builder.push_record(vec![format!(
+                            "{i:<3} Pure Arg          [{}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, ...]",
```

Comment:
Might be worth making a macro function or something for the spacing after `Pure Arg`

Could just be something like
```
arg_title("Pure Arg")
...
fn arg_title(x: impl std::fmt::Display) -> String {
  format!("{:<18}", x);
}
```

Reply:
If I come across a need to format something like this again I will invest in such a macro